src/Common/src/Interop/Windows/kernel32/Interop.GetFileAttributesEx.cs

```diff
@@ -68,10 +67,8 @@ internal unsafe struct WIN32_FIND_DATA
             internal uint nFileSizeLow;
             internal uint dwReserved0;
             internal uint dwReserved1;
-            [MarshalAs(UnmanagedType.ByValTStr, SizeConst = 260)]
-            internal string cFileName;
-            [MarshalAs(UnmanagedType.ByValTStr, SizeConst = 14)]
-            internal string cAlternateFileName;
+            internal FixedString.Size260 cFileName;
+            internal FixedString.Size14 cAlternateFileName;
```

Comment:
I think that the problem with the type that you have invented is that they are coded in tricky way and it is not easy to understand what they do.  `Span<T>` is solving the same problem that you are trying to solve with these types in general way.

If you would like to have something that looks pretty and that is safe as much as possible, you should use `Span<T>`. Add a small getter that converts the field to `Span<T>` or `ReadOnlySpan<T>` and have everything else coded as methods that operate on Span<T>.

```
    private fixed char _cFileName[260];

    internal ReadOnlySpan<char> cFileName
    {
        get
        {
            fixed (char* p = _cFileName)
                return new ReadOnlySpan<char>(p, 260);
        }
    }
```

We may want to add automatic conversions from fixed buffer field to Span/ReadOnlySpan to C# - at which point these small getters can go away.